### PR TITLE
Univariate Time Series using Facebook Prophet

### LINKED ISSUE #6552: [Machine Learning] Univariate Time Series using Facebook Prophet
#### Proposed title of article
[Machine Learning] Univariate Time Series using Facebook Prophet 

#### Proposed article introduction
A time series is a sequence of data points that occur in successive order over some period of time. A time series allows one to see what factors influence certain variables from period to period. Examples of time series data are weather records, economic indicators, and patient health evolution metrics.  Time series forecasting is the process of analyzing time series data using statistics and modeling to make predictions and inform strategic decision-making. Time series analysis involves developing models to gain an understanding of the data to understand the underlying causes.

In time series, we have univariate, bivariate, and multivariate modeling. Univariate statistics summarize only one variable at a time, bivariate statistics compare two variables and multivariate statistics compare more than two variables. In this tutorial, we will be focusing on univariate time series modeling. 

Facebook Prophet is a procedure for forecasting time series data based on an additive model where non-linear trends are fit with yearly, weekly, and daily seasonality, plus holiday effects. It works best with time series that have strong seasonal effects and several seasons of historical data. Prophet is robust to missing data and shifts in the trend, and typically handles outliers well.

#### Key takeaways 
1. Univariate vs Bivariate vs Multivariate time series modeling.
2. Examples of univariate data.
3. Installing Facebook Prophet.
4. Dataset preparation.
5. Building NYC Energy Demand time series model.

#### Article quality
This tutorial is unique because we will compare the different types of times series: Univariate vs Bivariate vs Multivariate time series modeling. This will give a reader a detailed understanding before we focus on the univariate model. We then perform the analysis of univariate data to describe the data and find patterns that exist within it. We will then build a custom NYC Energy Demand time series model. This tutorial is detailed and step-by-step implementation.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by francis966**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6552

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Kayere** [comment]
Plagiarism check done ✅

**Comment by wilkistermumbi** [comment]
@francis966 Please perform a Grammarly check on your PR

**Comment by francis966** [comment]
Hello, @wilkistermumbi, I have used Used Grammarly. I have also used Hemingway Editor for readability.
Thanks.

grammarly-score

File: content/articles/univariate-time-series-using-facebook-prophet/index.md [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+pd.concat([df.set_index('ds')['y'],prediction.set_index('ds')['yhat']],axis=1).plot()
+```
+The code produces the following line chart:
+
+![y vs yhat line chart](/engineering-education/univariate-time-series-using-facebook-prophet/y-vs-yhat.jpg)
+
+From the output above:
+The blue line shows the actual values (y).
+ The orange line shows the predicted values (yhat). 
+The chart also shows the forecast values for the next 1000 days.
```
Comment (wilkistermumbi) [inline content/articles/univariate-time-series-using-facebook-prophet/index.md:335]:
```suggestion
From the output above:
- The blue line shows the actual values (y).
- The orange line shows the predicted values (yhat). 
- The chart also shows the forecast values for the next 1000 days. 
```

File: content/articles/univariate-time-series-using-facebook-prophet/index.md [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+- `yhat`: It contains the prediction/forecast values of the time series model.
+- `yhat_lower`: It contains the lower bound of the prediction/forecast values.
+- `yhat_upper`: It contains the upper bound of the prediction/forecast values.
+
+The prediction values:
+
+![Predicted values](/engineering-education/univariate-time-series-using-facebook-prophet/predicted-values-tail.jpg)
+
+#### First five rows of the prediction results
+Use this code:
```
Comment (wilkistermumbi) [inline content/articles/univariate-time-series-using-facebook-prophet/index.md:313]:
```suggestion
Use this code to see the first five rows of the prediction results:
```

File: content/articles/univariate-time-series-using-facebook-prophet/index.md [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+df.head()
+```
+The output:
+
+![First five rows](/engineering-education/univariate-time-series-using-facebook-prophet/first-rows.jpg)
+
+To see the last five rows of our dataset, use this code:
+
+```python
+df.head()
```
Comment (wilkistermumbi) [inline content/articles/univariate-time-series-using-facebook-prophet/index.md:248]: This should be `df.tail()`

**Comment by wilkistermumbi** [comment]
@francis966 Hi. This is a nice read. Please work on the minor corrections before we can move on to final review.

**Comment by francis966** [comment]
Hello @wilkistermumbi, I have worked on the minor corrections.
Thanks

**Comment by wilkistermumbi** [comment]
@francis966 Please check my final commits. Do not confuse `df.tail()` with `df.head().`

Otherwise, this PR will proceed for final review.

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**